### PR TITLE
BUG: `DataFrame.resample` changing index type to `MultiIndex` when dataframe is empty and upsampling

### LINKED ISSUE #55572: BUG: When upsampling an empty dataframe, the index type changes from DatetimeIndex to MultiIndex
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame([], index=pd.DatetimeIndex([]))

# this returns a MultiIndex rather than a DatetimeIndex
df.resample("D").ffill().index
```

### Issue Description

When upsampling an empty dataframe, the index type changes from `DatetimeIndex` to `MultiIndex`.

This applies to all the upsampling methods, i.e. `resample` followed by `fillna`, `ffill`, `bfill`, `nearest`, `asfreq`, or `interpolate`.

This only happens on empty dataframes or series, and only when upsampling (for example `resample("D").mean()` works as expected).

This regression was introduced in pandas 2.0.0

### Expected Behavior

The resampled dataframe should have a `DatetimeIndex` rather than a `MultiIndex`.

### Installed Versions

pandas              : 2.1.1
numpy               : 1.26.0
pytz                : 2023.3.post1
dateutil            : 2.8.2
setuptools          : 68.2.2
pip                 : 23.3
tzdata              : 2023.3

all others None

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Closes #55572
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit)
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.3.rst` file

### CODE REVIEW

File: doc/source/whatsnew/v2.1.3.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
+- Bug in :meth:`DataFrame.resample` changing index type to :class:`MultiIndex` when the dataframe is empty and using an upsample method (:issue:`55572`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.3.rst:23]: Could you move this to 2.1.4?

Comment (Charlie-XIAO) [reply doc/source/whatsnew/v2.1.3.rst:23]: Sorry for the delay, will do it rn.

**Comment by Charlie-XIAO** [comment]
@mroeschke Changelog updated.

File: doc/source/whatsnew/v2.1.4.rst [hunk 2]
```diff
@@ -23,6 +23,7 @@ Bug fixes
 - Bug in :class:`Series` constructor raising DeprecationWarning when ``index`` is a list of :class:`Series` (:issue:`55228`)
 - Bug in :class:`Series` when trying to cast date-like string inputs to :class:`ArrowDtype` of ``pyarrow.timestamp`` (:issue:`56266`)
 - Bug in :class:`Timestamp` construction with ``ts_input="now"`` or ``ts_input="today"`` giving a different unit from :meth:`Timestamp.now` or :meth:`Timestamp.today` (:issue:`55879`)
+- Bug in :meth:`DataFrame.resample` changing index type to :class:`MultiIndex` when the dataframe is empty and using an upsample method (:issue:`55572`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.4.rst:26]: Sorry could you move again to 2.2? (2.1.4 was released last week)

Comment (Charlie-XIAO) [reply doc/source/whatsnew/v2.1.4.rst:26]: Sure.

**Comment by Charlie-XIAO** [comment]
@mroeschke Sorry for the delay. Changelog updated now.

**Comment by mroeschke** [comment]
So sorry this keeps getting missed. Could you please the whatsnew note to `2.2.1.rst`? I can merge as soon as that's made

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**